core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -297,6 +306,7 @@ private[spark] class TaskSchedulerImpl(
       taskSetsForStage -= manager.taskSet.stageAttemptId
       if (taskSetsForStage.isEmpty) {
         taskSetsByStageIdAndAttempt -= manager.taskSet.stageId
+        stageIdToFinishedPartitions -= manager.taskSet.stageId
```

Comment:
Should we also add this code when calling `killTasks`?

Reply:
I didn't find method `killTasks` in TaskSchedulerImpl, and for some similar func, e.g. `cancelTasks`, I think it's unnecessary.